File: app/src/test/java/com/fractalwrench/crazycats/common/PreConditionsTest.java
package com.fractalwrench.crazycats.common;

import org.junit.Test;

import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;

public class PreConditionsTest {

    @Test(expected = InvocationTargetException.class)
    public void checkNonInstantiation() throws Exception {
        Constructor<?> constructor = PreConditions.class.getDeclaredConstructors()[0];
        constructor.setAccessible(true);
        constructor.newInstance();
    }

    @Test
    public void checkNotNull() throws Exception {
        PreConditions.checkNonNull("");
    }

    @Test(expected = IllegalArgumentException.class)
    public void checkNull() throws Exception {
        PreConditions.checkNonNull(null);
    }

}

File: app/src/test/java/com/fractalwrench/crazycats/common/PreConditionsTest.kt
package com.fractalwrench.crazycats.common

import org.junit.Test
import java.lang.reflect.InvocationTargetException

class PreConditionsTest {

    @Test(expected = InvocationTargetException::class)
    @Throws(Exception::class)
    fun checkNonInstantiation() {
        val constructor = PreConditions::class.java.declaredConstructors[0]
        constructor.isAccessible = true
        constructor.newInstance()
    }

    @Test
    @Throws(Exception::class)
    fun checkNotNull() {
        PreConditions.checkNonNull("")
    }

    @Test(expected = IllegalArgumentException::class)
    @Throws(Exception::class)
    fun checkNull() {
        PreConditions.checkNonNull(null)
    }

}